packages/graph/hash_graph/lib/graph/src/api/web/mod.rs

```diff
@@ -0,0 +1,99 @@
+//! The Axum webserver for accessing the Graph API datastore operations.
+//!
+//! Handler methods are grouped by routes that make up the `web_api`. See [`data_type`] for a
+//! grouped router example
+
+mod api_resource;
+mod data_type;
+mod error;
+mod property_type;
+
+use axum::{routing::get, Extension, Json, Router};
+use utoipa::{openapi, Modify, OpenApi};
+
+use self::api_resource::RoutedResource;
+use crate::datastore::Datastore;
+
+fn api_resources<T: Datastore>() -> Vec<Router> {
```

Comment:
Does this need to be generic over the Datastore? In my head we wouldn't have the external facing API dependent on the datastore implementation at all. (Especially if we want to mix and match datastores)

Reply:
We can make a `CombinedDatastore` that `impl`s `Datastore` which uses ones underneath so I actually think this is fine, nevermind.

Reply:
We talked about this off-thread. The conclusion is that we do want to be generic over the `Datastore` to _some degree_ but not as it is implemented currently. We'll keep it as is for now, and not optimize for an unknown case.

The unknown case is mixing different `Datastore` implementors for different storage tech, which we may want to do at some point.